python/lib/core/dmod/test/test_context.py

```diff
@@ -0,0 +1,1599 @@
+
+    Returns:
+        An integer representing the size of passed in values
+    """
+    total: int = 0
+
+    for item in args:
+        if isinstance(item, typing.Sized):
+            total += len(item)
+        else:
+            total += item
+
+    return total
+
+
+def make_word(min_length: int = None, max_length: int = None, character_set: str = None, avoid: str = None) -> str:
+    """
+    Create a random jumble of characters to build a new word
+
+    Args:
+        min_length: The shortest the word can be
+        max_length: The longest a word can be
+        character_set: What characters can make up the word
+        avoid: A word to avoid creating
+
+    Returns:
+        A semi random string of a semi-random length
+    """
+    if min_length is None:
+        min_length = 2
+
+    if max_length is None:
+        max_length = 8
+
+    max_length = max(5, max_length)
+
+    if character_set is None:
+        character_set = string.ascii_letters + string.digits
+
+    word: str = avoid
+
+    while word == avoid:
+        word = ''.join(random.choice(character_set) for _ in range(random.randint(min_length, max_length)))
+
+    return word
+
+
+def make_number(minimum: int = 0, maximum: int = 3000, avoid: int = None) -> int:
+    """
+    Create a random number
+
+    Args:
+        minimum: The minimum allowable number
+        maximum: The maximum allowable number
+        avoid: A number to avoid
+
+    Returns:
+        A random number
+    """
+    number = random.randint(minimum, maximum)
+
+    while number == avoid:
+        number = random.randint(minimum, maximum)
+
+    return number
+
+
+def make_numbers(
+    minimum: int = 0,
+    maximum: int = 3000,
+    length: int = None,
+    avoid: typing.Sequence[int] = None
+) -> typing.Tuple[int, ...]:
+    """
+    Make a tuple of random numbers
+
+    Args:
+        minimum: the minimum value of the numbers to generate
+        maximum: The maximum value of the numbers to generate
+        length: The length of the sequence of numbers to generate
+        avoid: A sequence of numbers to avoid generating
+
+    Returns:
+        A tuple of random integers
+    """
+    if length is None:
+        length = random.randint(4, 12)
+
+    numbers = tuple(
+        make_number(minimum=minimum, maximum=maximum)
+        for _ in range(length)
+    )
+
+    while numbers == tuple(avoid):
+        numbers = tuple(
+            make_number(minimum=minimum, maximum=maximum)
+            for _ in range(length)
+        )
+
+    return numbers
+
+
+class Sentinel:
+    """
+    Represents a value that represents a void of anything. This differs from 'None' in that 'None' is an
+    acceptable value. Encountering a `Sentinel` value indicates that something went wrong.
+    """
+    def __eq__(self, other):
+        return False
+
+    def __hash__(self):
+        return -0
+
+    def __bool__(self):
+        return False
+
+
+SENTINEL = Sentinel()
```

Comment:
Why not just use `object()`. This is the standard way to create a sentinel value.

Reply:
Used for lines like `if self.ignore_result and isinstance(expectation, Sentinel):`. Can't do `isinstance(expectation, Object)` or `expectation == object()` and yield the same types of results.